### PR TITLE
Update documentation and samples for external_pem rename

### PR DESCRIPTION
**Pull request by Lekensteyn**
While at it, try to make the README more helpful for a new user.

Hold on for a moment, the plugin architecture looks nice, but I am not sure if this should the way forward for external plugins. For example, changing #42 could also influence external plugins (its interface is not well-defined).

It is shown that the names also do not stay consistent (external_pem.sh -> external.sh). What if a user wants to execute a .py script (external.py) script instead? Currently it would use the external.sh name...

There is also no way to signal from an external plugin that it stores a single key, but is able to accept other files too (certs, etc.).

### CODE REVIEW

File: simp_le.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ class IOPlugin(object):
     Unless otherwise stated, plugin data components are typically
     filled with the following data:
 
-    - for `account_key`: private accoutn key, an instance of `acme.jose.JWK`
```
Comment (kuba) [inline simp_le.py:241]: i'd prefer if typo fixes would be in separate commits or even PRs, thanks

Comment (kuba) [inline simp_le.py:241]: that being said:
- there is another PR that covers it
- you can keep it here

File: examples/external.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
-# Dummy example external script that loads/saves key/cert/chain to
-# /tmp/foo; `simp_le -f external_pem.sh`.
+# Dummy example external script that loads/saves (account) key/cert/chain to
```
Comment (kuba) [inline examples/external.sh:2]: Can you make it explicit? `account_key/key/cert/chain`

File: README.rst [hunk 3]
```diff
@@ -58,7 +63,7 @@ Manifest
     -  ``1`` if renewal not necessary;
     -  ``2`` in case of errors.
 
-12. ``--server`` (support multiple CAs).
+12. ``--server`` (support non-official ACME servers for custom CAs).
```
Comment (kuba) [inline README.rst:66]: there is no "official" ACME server; please revert this change or change wording; it's so happens to be that Let's Encrypt is a CA from the same folks that design the protocol

File: README.rst [hunk 4]
```diff
@@ -12,6 +12,11 @@ Simple `Let’s Encrypt`_ client.
       -d example.com -d www.example.com --default_root /var/www/html \
       -d example.net:/var/www/other_html
 
+This creates a new key file and requests a single certificate for
+``example.com`` and ``example.net``. The key files and certificates are
+stored in the current working directory. If no ``account_key.json`` file
+is found, a new account is registered using ``you@example.com``.
```
Comment (kuba) [inline README.rst:18]: I personally find this paragraph a bit misleading as it doesn't show the true nature of `simp_le`: i.e. the idempotent behavior (creationg of new files only if necessary, etc.).

**Comment by kuba** [comment]
I've updated references to `external_pem.sh` in 5211a83fc60e1bf4e09f0e7a443f3bf8b39a0405.

**Comment by Lekensteyn** [comment]
The external_pem thing docs and file are fixed/renamed, closing this PR. Not sure about the state of the external plugin, I saw some changes on testing that interface :+1:
